### PR TITLE
11 integrate login signup

### PR DESCRIPTION
**Pull request by nicholasrobinharrison**
**Done**
-Created simple login, signup requests. 
-Added error handling to login. 
-Added an error message pop up to the reusable components, that I didn't end up using, but it may come in handy later so I won't delete it for now
-JWT stored locally
-user passed into user context, working nicely
-added a placeholder homepage
-redirecting to homepage on successful login

### CODE REVIEW

File: client/src/pages/Signup.js [hunk 1]
```diff
@@ -64,6 +58,25 @@ const Signup = () => {
   };
 
   const submit = () => {
+    let user = {
+      email: email,
+      name: name,
+      password: password,
+      password2: secondPassword
```
Comment (ashan36) [inline client/src/pages/Signup.js]: We changed this field name to "confirmPassword" on the backend.

File: client/src/pages/Login.js [hunk 2]
```diff
@@ -36,9 +36,73 @@ const Login = () => {
...
+    if (passwordError) {
+      setPasswordError(null);
+    }
+  }, [password]);
 
   const submit = () => {
-    return; // placeholder
+    async function login(user) {
+      try {
+        let res = await fetch("/login", {
```
Comment (skassam21) [inline client/src/pages/Login.js]: let's favour const over let. Only use let if you actually need to re-bind a variable

File: client/src/components/ErrorMessage.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  }
+});
+
+const ErrorMessage = props => {
+  const classes = useStyles();
+
+  return (
+    <Box className={classes.container}>
+      <Typography className={classes.text}>{props.message}</Typography>
+    </Box>
```
Comment (skassam21) [inline client/src/components/ErrorMessage.js:28]: I like that you are creating re-usable components! Looking forward to how this looks :) Wondering how it'll compare to using a snackbar component instead

File: client/src/pages/Signup.js (hunk 1 above)
Comment (nicholasrobinharrison) [reply client/src/pages/Signup.js]: ok, i'll make a note to update this once i pull in the backend changes

File: client/src/pages/Login.js (hunk 2 above)
Comment (nicholasrobinharrison) [reply client/src/pages/Login.js]: got it

File: client/src/components/ErrorMessage.js (hunk 3 above)
Comment (nicholasrobinharrison) [reply client/src/components/ErrorMessage.js:28]: I tried using snackbars initially but I couldn't quite figure it out. From the example on the material ui page they were using an alert from @ material-ui/lab, but it didn't work for me.

File: client/src/pages/Signup.js [hunk 4]
```diff
@@ -64,68 +61,102 @@ const Signup = () => {
...
+          method: "post",
+          headers: {
+            "Content-Type": "application/json"
+          },
+          body: JSON.stringify(user)
+        });
+        let json = await res.json();
+        console.log(json);
+        if (json.errors) {
+          console.log(json.errors[0]);
```
Comment (skassam21) [inline client/src/pages/Signup.js]: we will want to eventually show these errors as snackbars, can be done in a future PR though

File: client/src/components/ErrorMessage.js (hunk 3 above)
Comment (skassam21) [reply client/src/components/ErrorMessage.js:28]: for alerts you'd have to install @material-ui/lab with npm, but I don't think we need to go that right. You may want to play around more with snackbars to see how they work, they will be very useful

**Review by ashan36 — APPROVED** [review]
Nice job

**Pull request merged**